### PR TITLE
Adding frame protector to S2A/Envoy Integration

### PR DESCRIPTION
**Pull request by dmbm0830**

### CODE REVIEW

File: source/extensions/transport_sockets/s2a/tsi_frame_protector.cc [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  grpc_slice_buffer_destroy(&protected_buffer);
+  grpc_slice_buffer_destroy(&unprotected_buffer);
+
+  return TSI_OK;
+}
+
+} // namespace S2A
+} // namespace TransportSockets
+} // namespace Extensions
+} // namespace Envoy
```
Comment (tavishvaidya) [inline source/extensions/transport_sockets/s2a/tsi_frame_protector.cc]: Nit: Please add a newline at the end.

File: source/extensions/transport_sockets/s2a/tsi_frame_protector.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+private:
+  CFrameProtectorPtr frame_protector_;
+};
+
+using TsiFrameProtectorPtr = std::unique_ptr<TsiFrameProtector>;
+
+} // namespace S2A
+} // namespace TransportSockets
+} // namespace Extensions
+} // namespace Envoy
```
Comment (tavishvaidya) [inline source/extensions/transport_sockets/s2a/tsi_frame_protector.h:50]: Nit: Please add a newline here as well.

**Review by tavishvaidya — COMMENTED** [review]
How about updating the commit message to something like "Adds the frame protector to S2A transport socket extension" or something similar?

File: source/extensions/transport_sockets/s2a/tsi_frame_protector.h (hunk 2 above)
Comment (dmbm0830) [reply source/extensions/transport_sockets/s2a/tsi_frame_protector.h:50]: There is already a newline at the end of this file.

Comment (tavishvaidya) [reply source/extensions/transport_sockets/s2a/tsi_frame_protector.h:50]: Ah, I see. Thanks!

File: source/extensions/transport_sockets/s2a/tsi_frame_protector.cc [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+#include "source/common/buffer/buffer_impl.h"
+#include "source/common/common/assert.h"
+
+#include "src/core/tsi/transport_security_grpc.h"
+#include "src/core/tsi/transport_security_interface.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace S2A {
```
Comment (tavishvaidya) [inline source/extensions/transport_sockets/s2a/tsi_frame_protector.cc:12]: @asraa Are there any Envoy guidelines around this being "S2A" or "S2a"? "S2A" is definitely preferable.

**Review by tavishvaidya — APPROVED** [review]
LGTM! Left a comment for Asra, but we can address that separately without blocking this PR.

**Pull request merged**